eos/form-factors/parametric-bsz2015.hh

```diff
@@ -186,6 +188,66 @@ namespace eos
     extern template class BSZ2015FormFactors<BsToPhi, PToV>;
     extern template class BSZ2015FormFactors<BsToDsstar, PToV>;
 
+
+    // P -> P
+    template <typename Process_>
+    class BSZ2015FormFactorTraits<Process_, PToP> :
+        public virtual ParameterUser
+    {
+        public:
+            // The following parameters are part of the parameterization and should match the
+            // the ones used for the extraction of the coefficients of the z-expension
+            UsedParameter m_B, m_P;
+            UsedParameter m_R_0p, m_R_1m;
+
+            const double m_R2_0p, m_R2_1m;
+            const double tp, tm, t0;
+
+            static const std::map<std::tuple<QuarkFlavor, QuarkFlavor>, std::string> resonance_0p_names;
+            static const std::map<std::tuple<QuarkFlavor, QuarkFlavor>, std::string> resonance_1m_names;
+
+            BSZ2015FormFactorTraits(const Parameters & p) :
+                m_B(UsedParameter(p[std::string(Process_::name_B) + "@BSZ2015"], *this)),
+                m_P(UsedParameter(p[std::string(Process_::name_P) + "@BSZ2015"], *this)),
+                m_R_0p(UsedParameter(p[resonance_0p_names.at(Process_::partonic_transition)], *this)),
+                m_R_1m(UsedParameter(p[resonance_1m_names.at(Process_::partonic_transition)], *this)),
+                m_R2_0p(power_of<2>(m_R_0p)),
+                m_R2_1m(power_of<2>(m_R_1m)),
+                tp(power_of<2>(m_B + m_P)),
+                tm(power_of<2>(m_B - m_P)),
+                t0(tp * (1.0 - std::sqrt(1.0 - tm / tp)))
+            {
+            }
+
+            complex<double> calc_z(const complex<double> & s) const
+            {
+                return (std::sqrt(tp - s) - std::sqrt(tp - t0)) / (std::sqrt(tp - s) + std::sqrt(tp - t0));
+            }
+
+            double calc_z(const double & s) const
+            {
+                return real(calc_z(complex<double>(s, 0.0)));
+            }
+    };
+
+    template <typename Process_>
+    const std::map<std::tuple<QuarkFlavor, QuarkFlavor>, std::string>
+    BSZ2015FormFactorTraits<Process_, PToP>::resonance_0p_names
+    {
+        { std::make_tuple(QuarkFlavor::bottom, QuarkFlavor::down), "mass::B_d,0@BSZ2015" },
```

Comment:
That naming is weird. We call the ground-state mass ``mass::B_d`` elsewhere, not ``mass::B_d,0``. I think the parameters should be identically named, except for the suffix ``@BSZ2015``.

Reply:
Ah no, this is the 0p excited meson!